tests/testthat/helper-mcstate2.R

```diff
@@ -34,12 +34,29 @@ ex_dust_sir <- function(n_particles = 100, n_threads = 1,
   model <- sir$new(list(), 0, n_particles, seed = 1L, n_threads = n_threads,
                    deterministic = deterministic)
   model$set_data(dust::dust_data(dat))
+  model$set_index(c(2, 4))
 
   prior_beta_shape <- 1
   prior_beta_rate <- 1 / 0.5
   prior_gamma_shape <- 1
   prior_gamma_rate <- 1 / 0.5
 
+  trajectories <- NULL
+
+  ## In the new dust wrapper we'll need to make this nicer; I think
+  ## that this is pretty painful atm because we wrap via the particle
+  ## filter method in mcstate1.  This version replicates most of what
+  ## we need though, which is some subset of the model
+  details <- function(i, j) {
+    if (save_trajectories) {
+      traj <- trajectories[, i, , drop = FALSE]
+      dim(traj) <- dim(traj)[-2]
```

Comment:
Maybe some comments here on what structure `trajectories` is, and what those two lines do / why ? I am staring at it a bit...!

Perhaps better names for i and j - and why don't we use/need j?